### PR TITLE
Replace KeeperMinimumRequiredConfirmation with global MinIncomingConfirmations

### PR DESCRIPTION
**Pull request by RensR**
add KeeperSpec.Confirmations to overwrite global MinIncomingConfirmations when needed.

### CODE REVIEW

File: core/store/migrate/migrations/0083_add_keeper_confirmations_to_spec.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-- +goose Up
+ALTER TABLE keeper_specs
+    ADD COLUMN confirmations integer;
```
Comment (samsondav) [inline core/store/migrate/migrations/0083_add_keeper_confirmations_to_spec.sql:3]: Do we want to call it `min_required_incoming_confirmations` just to keep it consistent?

File: core/testdata/testspecs/v2_specs.go [hunk 2]
```diff
@@ -124,6 +124,7 @@ type KeeperSpecParams struct {
 	ContractAddress string
 	FromAddress     string
 	EvmChainID      int
+	Confirmations   int
```
Comment (samsondav) [inline core/testdata/testspecs/v2_specs.go:127]: Is it incoming or outgoing?

`MinRequiredIncomingConfirmations` or `MinIncomingConfirmations` would be clearer

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/store/migrate/migrations/0083_add_keeper_confirmations_to_spec.sql (hunk 1 above)
Comment (RensR) [reply core/store/migrate/migrations/0083_add_keeper_confirmations_to_spec.sql:3]: I used the same name as in the VRF spec, should we change that one too?

File: core/services/keeper/registry_synchronizer_core.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ type RegistrySynchronizerOptions struct {
 	JRM                 job.ORM
 	LogBroadcaster      log.Broadcaster
 	SyncInterval        time.Duration
-	MinConfirmations    uint64
+	MinConfirmations    uint32
```
Comment (samsondav) [inline core/services/keeper/registry_synchronizer_core.go:37]: Incoming or outgoing? Please be clear

File: core/store/migrate/migrations/0083_add_keeper_confirmations_to_spec.sql (hunk 1 above)
Comment (samsondav) [reply core/store/migrate/migrations/0083_add_keeper_confirmations_to_spec.sql]: Yes, to be honest. "confirmations" is ambiguous

**Review by samsondav — APPROVED** [review]

**Pull request merged**